### DISCUSSION TITLE
Has anyone tried (or thought about) running this on Azure App Service?

### DISCUSSION THREAD

**Discussion by BJReplay** [jasonacox/Powerwall-Dashboard#107] [Q&A] [answered]
I don't have an appropriate host to run it on at home.

I do have an always free Azure App Service subscription available (including a VPN back to my router, so my Powerwall can appear to be local).

I should - in theory - be able to set up everything needed on an Azure App Service, but before I started, I wanted to check in to see if anyone else had thought about this and ruled it out, or tried it, and failed, or better still, tried it and succeeded.

**Comment by jasonacox** [comment]
I love this idea @BJReplay !  It would be good to add this as an alternative setup path for anyone without a local host they can use for the Dashboard.  Can you give us specifics on your setup and how you got this free subscription that has the VPN back?

**Comment by BJReplay** [comment]
I haven't managed to get it going, yet, but will have a go, and if I get it up and running, will do my best to write it up.

I did try to get it up and running on WSL, but couldn't get docker running, and have actual work to do rather than this, so haven't come back to it yet.

I tend to do things via the admin GUI rather than attempt to script using PowerShell, so if I get it running, the specifics won't be a nice PS script.

If you go to https://azure.microsoft.com/en-us/pricing/free-services/ you can see what's free.  The two things you need are https://azure.microsoft.com/en-us/products/virtual-network/ (so that when you get the Powerwall Dashboard running, the IP address is a local address, via the VPN) and https://azure.microsoft.com/en-us/products/app-service/ and it is the app services that I think will provide the other prerequisites for Powerwall Dashboard.

To get there, you sign up for a free trial for a month, and then convert it to a pay as you go service at the free monthly plan tier.  Provided you don't set up any services that cost actual money, it remains free.

I've got other services running on the aways free tier, including some other live streaming of Powerwall data, monitoring of FitBit data, and other things like that, and have previously set up a VPN from my router to Azure as a proof of concept.

**Comment by BJReplay** [comment]
Well, it can't be done - at least not yet.

See - https://learn.microsoft.com/en-us/azure/app-service/configure-custom-container?pivots=container-linux#configure-multi-container-apps

Limitations:

- VNET integration is not supported for Docker Compose scenarios

Unsupported docker compose YAML:
- [depends_on](https://learn.microsoft.com/en-us/azure/app-service/faq-app-service-linux#how-do-i-use-depends-on-) (ignored)
- ports other than 80 and 8080 (ignored) - this may not be a problem for inter-contain comms, but hard to say.

The VNET was going to be critical for communication back to the Powerwall at home.

**Comment by BJReplay** [comment, marked as answer]
Two months later, answering my own question - this is possible...

- Oracle Cloud for the cloud hosting (https://oraclecloud.com/) supports beefy ARM instances (4VCPU, 24GB RAM in their always free configuration).  I'm running my own mastodon instance on this.
- Tailscale (https://tailscale.com/) supports point to point VPN and subnet routing (so you can access the powerwall - though is a little tricky to get going) over Wireguard, and I highly recommend trying it out.  It includes clients for Windows, IOS, Android, Mac, pretty much any flavour of Linux, synology NAS, and a bunch of pre-built linux images, including setting up on OpenWRT (https://openwrt.org/docs/guide-user/services/vpn/tailscale/start) to create a tunnel from home to cloud to enable the cloud to talk to the Powerwall gateway at home.

I ran this up as a quick Proof of Concept on an existing cloud instance that I had already running docker; I wasn't running docker in non-sudo mode, so had to manually set it up (i.e. script up the pypowerwall conf, then sudo compose-dash.sh up -d, but got it up and running long enough to get it up.

I wouldn't recommend it over buying a Raspberry Pi, but it is doable.

**Reply by jasonacox** [reply]
Thanks @BJReplay !  This is a rough diagram, but does this capture the basic setup for anyone wanting to duplicate what you did? 

<img width="813" alt="image" src="https://user-images.githubusercontent.com/836718/210706629-f57655fc-8f03-4318-932f-e80e5bf3582f.png">

**Reply by BJReplay** [reply]
Yes, it does - the Tailscale client is also running on the ARM VM in the Oracle Cloud - and the Mac / Windows workstation can also be an OpenWRT router or RaspberryPI or Synology NAS as well (all of which could run the dashboard, of course) - but the potential advantage is that you can get the Tailscale VPN client running on an existing OpenWRT router.

It means that you can also have your dashboard exposed via something like nginx proxy manager and available publicly via a reverse proxy.

**Reply by jasonacox** [reply]
Thanks @BJReplay  - so more like this?

<img width="807" alt="image" src="https://user-images.githubusercontent.com/836718/210918876-62962f58-a68a-4143-ae93-7f0b8d9114ef.png">

**Comment by BJReplay** [comment]
Yep, that's it!
